### PR TITLE
Deprecate Helm v2 SDKs

### LINKED ISSUE #1342: Deprecate Helm v2 support
### Problem description

[Helm v2 is EOL](https://helm.sh/blog/helm-v2-deprecation-timeline/), so we should deprecate our support for it in preparation for its later removal. Our Helm v3 support is already backward-compatible, so users should be able to upgrade without disruption to deployed resources.

We should also warn users that the `stable` and `incubator` repos [may stop working](https://github.com/helm/charts#deprecation-timeline) in the near future (November 13, 2020).

### PR DESCRIPTION
**Pull request by lblackstone**
### Related issues (optional)
Fix #1342

### CODE REVIEW

File: provider/pkg/gen/nodejs-templates/helm/v2/helm.ts [hunk 1]
```diff
@@ -143,6 +143,7 @@ export class Chart extends yaml.CollectionComponentResource {
         config: ChartOpts | LocalChartOpts,
         opts?: pulumi.ComponentResourceOptions
     ) {
+        pulumi.log.warn("Chart is deprecated: helm/v2/Chart is deprecated by helm/v3/Chart and will be removed in a future release.")
```
Comment (justinvp) [inline provider/pkg/gen/nodejs-templates/helm/v2/helm.ts:146]: We should also add `@deprecated` JSDoc tag to the bottom of the class's doc comment ([like this](https://github.com/pulumi/pulumi-aws/blob/381d01b575bb9f5b449b8f4992ed3145bd0d5ffa/sdk/nodejs/ses/confgurationSet.ts#L7-L9)). Which now has nice [IDE support in VS Code](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#-deprecated--support).

File: provider/pkg/gen/python-templates/helm/v2/helm.py [hunk 2]
```diff
@@ -153,6 +157,8 @@ def omit_resource(obj, opts):
         :param Optional[pulumi.ResourceOptions] opts: A bag of options that control this
                resource's behavior.
         """
+        warnings.warn("helm.v2.Chart has been deprecated in favor of helm.v3.Chart and will be removed in a future"
+                      " release.", DeprecationWarning)
```
Comment (justinvp) [inline provider/pkg/gen/python-templates/helm/v2/helm.py:161]: Is this multi-line string right?

**Review by justinvp — APPROVED** [review]
Should `nuget/Pulumi.Kubernetes.2.1.0-alpha.1601933518.nupkg` be deleted?

LGTM otherwise.

File: provider/pkg/gen/python-templates/helm/v2/helm.py (hunk 2 above)
Comment (lblackstone) [reply provider/pkg/gen/python-templates/helm/v2/helm.py]: Updated

**Pull request merged**